internal/commands/app/create.go

```diff
@@ -354,11 +342,102 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 	return nil
 }
 
+func (cmd CommandCreate) handleCreateTemplateApp(
+	profile *user.Profile,
+	ui terminal.UI,
+	clients cli.Clients,
+	groupID string,
+	rootDir string,
+	dsClusters []dataSourceCluster,
+	dsDatalakes []dataSourceDatalake,
+) error {
+	if cmd.inputs.DryRun {
+		logs := make([]terminal.Log, 0, len(dsClusters)+len(dsDatalakes)+2)
+		logs = append(logs, terminal.NewTextLog(fmt.Sprintf(
+			"A Realm app would be created at %s using the '%s' template",
+			rootDir,
+			cmd.inputs.Template,
+		)))
+
+		for _, cluster := range dsClusters {
+			logs = append(logs, terminal.NewTextLog("The cluster '%s' would be linked as data source '%s'", cluster.Config.ClusterName, cluster.Name))
+		}
+		for _, datalake := range dsDatalakes {
+			logs = append(logs, terminal.NewTextLog("The data lake '%s' would be linked as data source '%s'", datalake.Config.DatalakeName, datalake.Name))
+		}
+		logs = append(logs, terminal.NewFollowupLog("To create this app run", cmd.display(true)))
+		ui.Print(logs...)
+		return nil
+	}
+
+	if len(dsClusters) != 1 {
+		return errors.New("could not send app creation request due to unexpected number of clusters")
+	}
+	createAppMetadata := realm.AppMeta{
+		Location:        cmd.inputs.Location,
+		DeploymentModel: cmd.inputs.DeploymentModel,
+		Environment:     cmd.inputs.Environment,
+		Template:        cmd.inputs.Template,
+		DataSource:      dsClusters[0],
+	}
+
+	createTemplateApp := func() (realm.App, error) {
+		s := spinner.New(terminal.SpinnerCircles, 250*time.Millisecond)
+		s.Suffix = " Creating template app..."
+		s.Start()
+		defer s.Stop()
+
+		appRealm, err := clients.Realm.CreateApp(
+			groupID,
+			cmd.inputs.Name,
+			createAppMetadata,
+		)
+		if err != nil {
+			return realm.App{}, err
+		}
+		return appRealm, nil
+	}
+	appRealm, err := createTemplateApp()
+	if err != nil {
+		return err
+	}
+
```

Comment:
when showing a spinner, it's good to have a `Log` printed right afterwards to prevent the effect of shifting text. just use the past tense of the spinner text, so its `"Created template app"` here